python/ray/train/batch_predictor.py

```diff
@@ -170,7 +171,16 @@ def predict(
         ):
             predictor_kwargs["use_gpu"] = True
 
+        # For DLPredictor, we use numpy first for better dev API as well as
+        # improved performance of zero-copy data conversion
+        batch_format = "pandas"
         ctx = DatasetContext.get_current()
+        if issubclass(predictor_cls, DLPredictor):
```

Comment:
This is breaking abstraction boundaries. We should not check predictor. class identity within batch predictor.